From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
Date: Sun, 17 Oct 2021 14:59:54 -0400
Subject: [PATCH] Undeprecate Delegate Generate Tree Method

This was wrongly deprecated because it was thought it did not include tile entities, however with some basic testing it certainly does support them.
The issue here is that the treetypes themselves don't use any generators that contain beehives, so you will never be able to encounter them via the api.

diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 4f673e9123145dc78564dc3eef0edf75795dafc2..9e8fcd18785ce0150f2a5a366fea4ae64243a64e 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -734,9 +734,8 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
      *     this method
      * @return true if the tree was created successfully, otherwise false
      * @see #generateTree(org.bukkit.Location, java.util.Random, org.bukkit.TreeType, org.bukkit.util.Consumer)
-     * @deprecated this method does not handle tile entities (bee nests)
      */
-    @Deprecated
+    // @Deprecated Paper - Remove incorrect deprecation
     public boolean generateTree(@NotNull Location loc, @NotNull TreeType type, @NotNull BlockChangeDelegate delegate);
 
     /**
